Example/Sources/View Controllers/AnnouncementExampleViewController.swift

```diff
@@ -0,0 +1,364 @@
+        guard conversationMessages.count > indexPath.row else { fatalError("message couldn't find") }
+
+        switch message.kind {
+        case .announcement:
+            return .black
+        default:
+            return isFromCurrentSender(message: message) ? .white : .black
+        }
+    }
+
+    func backgroundColor(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> UIColor {
+        guard conversationMessages.count > indexPath.row else { fatalError("message couldn't find") }
+
+        switch message.kind {
+        case .announcement:
+            return .lightGray
+        case .photo:
+            return .clear
+        default:
+            return isFromCurrentSender(message: message) ? UIColor.green : UIColor.lightGray
+        }
+    }
+
+    func messageStyle(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> MessageStyle {
+        guard conversationMessages.count > indexPath.row else { fatalError("message couldn't find") }
+        let conversationMessage = conversationMessages[indexPath.row]
+        if conversationMessage.isAnnouncement {
+            return .announcement
+        } else {
+            return  isFromCurrentSender(message: message) ? .rightBubble : .leftBubble
+        }
+    }
+
+    func configureAvatarView(_ avatarView: AvatarView, for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) {
+        guard !isFromCurrentSender(message: message) else { return }
+
+        guard conversationMessages.count > indexPath.row else { fatalError("message couldn't find") }
+        let conversationMessage = conversationMessages[indexPath.row]
+        guard shouldShowSenderInfo(for: message, at: indexPath), !conversationMessage.isAnnouncement else {
+            avatarView.isHidden = true
+            return
+        }
+        avatarView.isHidden = false
+
+        let avatar = Avatar(image: nil, initials: "S")
+        avatarView.set(avatar: avatar)
+    }
+
+    func didSelectMessage(at indexPath: IndexPath, message: MessageType) {
+        switch message.kind {
+        case .text:
+            break
+        case .photo(let media):
+            print("selected photo: \(media.size)")
+        case .announcement:
+            break
+        case .attributedText:
+            break
+        case .custom:
+            break
+        default:
+            break
+        }
+    }
+
+}
+
+extension AnnouncementExampleViewController: MessageCellDelegate {
+    func didTapMessage(in cell: MessageCollectionViewCell) {
+        guard let indexPath = messagesCollectionView.indexPath(for: cell),
+            let message = messagesCollectionView.messagesDataSource?.messageForItem(at: indexPath, in: messagesCollectionView) else { return }
+        print("didTapMessage \(message.messageId)")
```

Comment:
Still need debug prints?

Reply:
Ah, I want to leave these debug prints. Because these are examples.

Reply:
Thanks, got it :+1: